src/i8042.rs

```diff
@@ -24,15 +25,15 @@ const CMD_RESET_CPU: u8 = 0xFE;
 
 /// Errors encountered while handling i8042 operations.
 #[derive(Debug)]
-pub enum Error {
+pub enum Error<E> {
     /// Failed to trigger interrupt.
-    TriggerInterrupt(io::Error),
+    TriggerInterrupt(E),
```

Comment:
While this is the only error which can occur here, how about we don't define an `Error` enum, and simply return something like `Result<something, T::E>` from the methods that call `trigger` below?